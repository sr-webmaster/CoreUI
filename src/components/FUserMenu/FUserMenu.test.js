import Component from '@freshinup/core-ui/src/components/FUserMenu'
import { FIXTURE_USER } from '@freshinup/core-ui/tests/__data__/user'
import { mount } from '@vue/test-utils'

describe('UserMenu', () => {
  describe('Snapshots', () => {
    test('default', async () => {
      const wrapper = mount(Component, {
        propsData: {
          user: FIXTURE_USER,
          userIsAdmin: true
        }
      })
      expect(wrapper.find('.f-userMenu__userLevel').exists()).toBe(true)
      expect(wrapper.text()).toContain('Switch to Admin View')
      expect(wrapper.element).toMatchSnapshot()
    })

    test('consumerView false', async () => {
      const wrapper = mount(Component, {
        propsData: {
          user: FIXTURE_USER,
          userIsAdmin: true,
          consumerView: false
        }
      })
      expect(wrapper.text()).not.toContain('Switch to Admin View')
      expect(wrapper.element).toMatchSnapshot()
    })

    test('hideLevel true', async () => {
      const wrapper = mount(Component, {
        propsData: {
          user: FIXTURE_USER,
          userIsAdmin: true,
          hideLevel: true
        }
      })
      expect(wrapper.find('.f-userMenu__userLevel').exists()).toBe(false)
      expect(wrapper.element).toMatchSnapshot()
    })

    test('userField email', async () => {
      const wrapper = mount(Component, {
        propsData: {
          user: FIXTURE_USER,
          userIsAdmin: true,
          userField: 'email'
        }
      })
      expect(wrapper.text()).toContain(FIXTURE_USER.email)
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Visuals', () => {
    test('displays "Switch to Admin View" when not on /admin and user is Admin', () => {
      const wrapper = mount(Component, {
        propsData: {
          userIsAdmin: true,
          user: FIXTURE_USER
        },
        mocks: {
          $route: {
            path: '/users/4'
          }
        }
      })
      expect(wrapper.text()).toContain('Switch to Admin View')
      expect(wrapper.text()).not.toContain('Switch to User View')
    })
    test('displays "Switch to User View" when on /admin and user is Admin', () => {
      const wrapper = mount(Component, {
        propsData: {
          userIsAdmin: true,
          user: FIXTURE_USER
        },
        mocks: {
          $route: {
            path: '/admin/users'
          }
        }
      })
      expect(wrapper.text()).toContain('Switch to User View')
      expect(wrapper.text()).not.toContain('Switch to Admin View')
    })
    test('does not display "Switch to Admin View" when not on /admin and user is not Admin', () => {
      const wrapper = mount(Component, {
        propsData: {
          userIsAdmin: false,
          user: FIXTURE_USER
        },
        mocks: {
          $route: {
            path: '/users/4'
          }
        }
      })
      expect(wrapper.text()).not.toContain('Switch to User View')
      expect(wrapper.text()).not.toContain('Switch to Admin View')
    })
  })
})
