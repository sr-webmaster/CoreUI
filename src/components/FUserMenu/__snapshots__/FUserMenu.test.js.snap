// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu Snapshots consumerView false 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="layout pa-3 row"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar accent"
        style="height: 64px; width: 64px;"
      >
        <img
          alt="Demo Admin"
          src="https://i.pravatar.cc/300"
        />
      </div>
    </div>
     
    <div
      class="flex pl-3"
    >
      <h4
        class="title secondary--text"
      >
        
        Demo Admin
      
      </h4>
       
      <span
        class="v-chip mx-0 pl-3 pr-3 f-userMenu__userLevel v-chip--disabled v-chip--small theme--light secondary white--text"
        tabindex="-1"
      >
        <span
          class="v-chip__content"
        >
          
        Platform Admin
      
        </span>
      </span>
       
      <div>
        <!---->
        
          Laravel
        
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-list pb-0 theme--light"
    role="list"
  >
     
    <!---->
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__title"
        >
          Sign Out
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`UserMenu Snapshots default 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="layout pa-3 row"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar accent"
        style="height: 64px; width: 64px;"
      >
        <img
          alt="Demo Admin"
          src="https://i.pravatar.cc/300"
        />
      </div>
    </div>
     
    <div
      class="flex pl-3"
    >
      <h4
        class="title secondary--text"
      >
        
        Demo Admin
      
      </h4>
       
      <span
        class="v-chip mx-0 pl-3 pr-3 f-userMenu__userLevel v-chip--disabled v-chip--small theme--light secondary white--text"
        tabindex="-1"
      >
        <span
          class="v-chip__content"
        >
          
        Platform Admin
      
        </span>
      </span>
       
      <div>
        <!---->
        
          Laravel
        
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-list pb-0 theme--light"
    role="list"
  >
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
        href="/admin"
      >
        
        Switch to Admin View
      
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__title"
        >
          Sign Out
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`UserMenu Snapshots hideLevel true 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="layout pa-3 row"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar accent"
        style="height: 64px; width: 64px;"
      >
        <img
          alt="Demo Admin"
          src="https://i.pravatar.cc/300"
        />
      </div>
    </div>
     
    <div
      class="flex pl-3"
    >
      <h4
        class="title secondary--text"
      >
        
        Demo Admin
      
      </h4>
       
      <!---->
       
      <div>
        <!---->
        
          Laravel
        
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-list pb-0 theme--light"
    role="list"
  >
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
        href="/admin"
      >
        
        Switch to Admin View
      
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__title"
        >
          Sign Out
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`UserMenu Snapshots userField email 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="layout pa-3 row"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar accent"
        style="height: 64px; width: 64px;"
      >
        <img
          alt="Demo Admin"
          src="https://i.pravatar.cc/300"
        />
      </div>
    </div>
     
    <div
      class="flex pl-3"
    >
      <h4
        class="title secondary--text"
      >
        
        Demo Admin
      
      </h4>
       
      <span
        class="v-chip mx-0 pl-3 pr-3 f-userMenu__userLevel v-chip--disabled v-chip--small theme--light secondary white--text"
        tabindex="-1"
      >
        <span
          class="v-chip__content"
        >
          
        Platform Admin
      
        </span>
      </span>
       
      <div>
        
          demoAdmin@example.com
        
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-list pb-0 theme--light"
    role="list"
  >
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
        href="/admin"
      >
        
        Switch to Admin View
      
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__title"
        >
          Sign Out
        </div>
      </a>
    </div>
  </div>
</div>
`;
